src/components/Section.js

```diff
@@ -119,169 +160,158 @@ const Section = ({section, type}) => {
-          />
+    //check if sectionText needs to be rendered
+    if (item.sectionText != null) {
+      return (
+        <View style={styles.accordionContainer}>
+          <Text>{item.sectionText}</Text>
         </View>
-        <View>
+      );
+    }
+    //check if paragraphs exists
+    else if (paraFilter.length > 1) {
+      return (
+        <View style={styles.accordionContainer}>
+          <Text>
+            {item.subsectionText}
+            {'\n'}
+          </Text>
           <FlatList
             data={paraFilter}
-            keyExtractor={data => data.index}
-            renderItem={renderItemPara}
+            keyExtractor={item => item.field1}
+            listKey={(item2, index) => 'B' + index.toString()}
+            renderItem={renderParagraph}
           />
         </View>
-      </View>
-    );
+      );
+    } else {
+      return (
+        <View style={styles.accordionContainer}>
+          <Text>{item.subsectionText}</Text>
+        </View>
+      );
+    }
   };
 
-  const renderSubPara = ({item}) => {
+  const renderParagraph = ({item, index}) => {
+    let subParaData = dbData.filter(
+      (subParagraph, i) =>
+        dbData[i].paragraphLabel === item.paragraphLabel &&
+        dbData[i].paragraphText === item.paragraphText &&
+        dbData[i].subparagraphText !== null,
+    );
+
     return (
-      <View>
-        <Text style={styles.subParagraph}>
-          {item.subparagraph} {item.subParagraphText}
+      <View style={styles.paragraph}>
+        <Text>
+          {item.paragraphLabel} {item.paragraphText}
         </Text>
+        <FlatList
+          data={subParaData}
+          keyExtractor={item => item.field1}
+          listKey={(item3, index) => 'C' + index.toString()}
+          renderItem={renderSubParagraph}
+        />
       </View>
     );
   };
 
-  const renderItemPara = ({item}) => {
-    //console.log(item);
-
-    //filter out data that has sub paragraphs in order to pass to flatlist for sub paragraphs
-    let subparaData = dbData.filter(
-      (section, i) =>
-        dbData[i].paragraph === item.paragraph &&
-        dbData[i].paragraphText === item.paragraphText &&
-        dbData[i].subparagraph !== null,
+  const renderSubParagraph = ({item, index}) => {
+    return (
+      <View style={styles.subParagraph}>
+        <Text>
+          {item.subparagraphLabel} {item.subparagraphText}
+        </Text>
+      </View>
     );
-
-    //console.log(subparaData);
-
-    if (item.paragraph !== null) {
-      return (
-        <View>
-          <Text style={[styles.paragraph]}>
-            {/*  In{item.index} */}
-            {item.paragraph} {item.paragraphText}
-          </Text>
-          <FlatList
-            data={subparaData}
-            renderItem={renderSubPara}
-            keyExtractor={data => data.index}
-          />
-        </View>
-      );
-    }
-  };
-
-  const getItem = (data, index) => {
-    return data[index];
   };
 
   if (loading === true) {
```

Comment:
Shouldn't this be the other way around? If it's loading, then you shouldn't show anything; only once it's finished loading should you display the list.